### PR TITLE
Closes #389 - Campaign API Helper Classes

### LINKED ISSUE #389: Create and schedule a marketing campaign
With the [SendGrid Marketing Campaigns feature](https://sendgrid.com/docs/API_Reference/api_v3.html#campaigns) you can create a campaign, edit it, then schedule it for the future or for 'now'
Create a helper that uses the API to build the campaign
Then add the ability to build + schedule, build + send now, or only build the campaign
Please also add the ability to notify a specific email address or a list of addresses about the fact that the campaign was built, scheduled, or sent. This is especially helpful when developers and marketers are working together, so that notifications can be sent when an automation is run.

### PR DESCRIPTION
**Pull request by kylesalk**
# Fixes # 
Adds helper classes for campaigns

Closes #389

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
- Added helper classes for campaign, campaigns, and campaign scheduling
- Added utility functions for building, sending, and scheduling campaigns

If you have questions, please send an email to [SendGrid](mailto:dx@sendgrid.com), or file a GitHub Issue in this repository.

### CODE REVIEW

**Comment by kylesalk** [comment]
@thinkingserious - sorry about the opening/closing PRs. Not sure if there is a way to make commits after the PR, but the issues with earlier PRs should be fixed.

Hope the campaign helpers will be useful. I did have some issues with the schedule API (/campaigns/{campaign_id}/schedules). It follows the API with a {"send_at": timestamp}, but is getting  BAD REQUEST. Not sure how to troubleshoot...

File: test/test_campaigns.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            "list_ids": [2, 3],
+            "plain_content": "Check out our summer line!",
+            "segment_ids": [2, 3],
+            "sender_id": 4324,
+            "subject": "New Products for Summer!",
+            "title": "May Newsletter",
+            "suppression_group_id": 5123
+        }
+        camp = Campaign(**data)
+        self.assertDictEqual(camp.get(), data)
```
Comment (42B) [inline test/test_campaigns.py]: `self.assertEqual` takes care of this already?

(Just below this link)
* https://docs.python.org/3/library/unittest.html?highlight=assertequal#unittest.TestCase.addTypeEqualityFunc

> The list of type-specific methods automatically used by assertEqual() are summarized in the following table. Note that it’s usually not necessary to invoke these methods directly.

Method | Used to compare | New in
-- | -- | --
assertMultiLineEqual(a, b) | strings | 3.1
assertSequenceEqual(a, b) | sequences | 3.1
assertListEqual(a, b) | lists | 3.1
assertTupleEqual(a, b) | tuples | 3.1
assertSetEqual(a, b) | sets or frozensets | 3.1
assertDictEqual(a, b) | dicts | 3.1

File: test/test_campaigns.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        new_data = {
+            "categories": [
+                "fall line"
+            ],
+            "html_content": "<html><head><title></title></head><body><p>Check out our summer line!</p></body></html>",
+            "plain_content": "Check out our summer line!",
+            "subject": "New Products for Fall!",
+            "title": "August Newsletter"
+        }
+        self.assertDictEqual(camp.patch(**new_data), new_data)
```
Comment (42B) [inline test/test_campaigns.py]: Same as above, `self.assertEqual`?

File: test/test_campaigns.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+
+class TestScheduleObject(unittest.TestCase):
+
+    def test_components_to_timestamp(self):
+        schedule = Schedule(year=2018, month=12, day=1, hour=8, minute=23)
+        t_delta = datetime(2018, 12, 1, 8, 23) - datetime(1970, 1, 1)
+        unix_timestamp = t_delta.seconds + t_delta.days * 86400
+        self.assertEqual(schedule.timestamp, unix_timestamp)
+        self.assertDictEqual(schedule.get(), {"send_at": unix_timestamp})
```
Comment (42B) [inline test/test_campaigns.py]: `self.assertEqual`?

**Comment by thinkingserious** [comment]
Hello @kylesalk,<br /><br/>Thanks again for the PR!<br /><br />                                         You have earned 7 out of the 5 points needed to receive glorious SendGrid Hacktoberfest swag.<br /><br />                                         Please take a moment to checkout <a href="https://dx.sendgrid.com/hacktoberfest/issues">this link</a> to find more issues to get you past the required threshold or to simply continue the celebration.<br /><br />                                         Also, <strong>please be sure you have officially registered with us <a href="https://dx.sendgrid.com/hacktoberfest/join">here</a> by November 1, 2018</strong> to qualify.<br /><br />                                         If you have any questions you can email us at <a href="mailto:dx+hacktoberfest2018@sendgrid.com">dx+hacktoberfest2018@sendgrid.com</a>.<br /><br />                                         Thank you and Happy Hacktobering!<br /><br />                                         Team SendGrid

**Comment by childish-sambino** [comment]
Closing until PR comments have been addressed.

**Pull request closed without merging**